### DISCUSSION TITLE
Easiest way to create a page under another page's path?

### DISCUSSION THREAD

**Discussion by StagnationPoint** [requarks/wiki#3073] [Help / Questions] [answered]
I'm setting up a family wiki and want to give each member their own space, with permissions to prevent one member from messing with another member's pages. The group page rules in wiki.js look like this will work fine, just specify different page rules for each group by path.

I'm familiar with Confluence, so I figured the way this would work is to make a base page for each family member, then they could create everything they want under that. On Confluence, you'd navigate to your base page (e.g. "Joe's Page"), then click the Create button. The new page would automatically live under that path.

I'm not sure how to get the same ease of use with wiki.js. If I click to create a new page, it looks like you have to type in the path you want, and the field doesn't default to the page you're already on -- it defaults to "/new-page". Now, I can click "Joe's Page", which sets the base path, and then you still have to type "/new-page-name". This could be tough for little kids to figure out.

Am I missing something? Is there a "Create Subpage" button somewhere I'm not seeing, that lets you start editing a new page, then when you save it, sets the path under wherever you started, plus the page title?

Thanks.

**Comment by mattpr** [comment, marked as answer]
I very new to wiki.js too (3 days) but @NGPixel has been helpful/responsive so far and so trying to give back a little on the parts I have figured out.

You aren't missing something.  As far as I can tell there is no concept of sub pages.  Each page lives at a certain path.  As long as that path is unique and not colliding with some reserved prefixes then it is valid.  If a path has multiple slashes in it, those segments are treated as "virtual folders" for organizing content.

What you can have are pages that live at the following paths:

`/Mary`
`/Mary/marys-second-page`
`/Dave`
`/Dave/daves-second-page`

You could also have a page called `/one` and a page called `/one/two/three/four/five/six` and only have 2 pages with nothing in between.  If you later wanted to create a page called `/one/two/three` you could.

When you create pages with more than just the leading slash, you are creating "virtual folders".  These can be used when creating more pages to select (with mouse) WHERE you want the new page to be created.  So the key thing is make sure you have your virtual folders pre-created to make things less confusing for your users.  You can't actually create folders (they are virtual) so you need to create a page "below" that folder to imply the folder exists.  In the example above, creating a page `/one/two/three/four/five/six` would create one page that is nested in a series of virtual folders `/one/two/three/four/five/`.

I give 2 concrete examples below with screenshots.

The other thing... check your `settings > navigation`.  There are different modes.  The prettier mode involves manually managing the navigation which could be a good option if you have a limited set of users/pages.  However using the tree mode will make it easier to see the full "structure" of the wiki and might reveal some things you were missing while you are learning to use the wiki.

![image](https://user-images.githubusercontent.com/1196252/109031065-a80ca280-76c4-11eb-89d7-b66d75b0ee7f.png)


# Example 1: page-first

Here is `/Dave` page.

![image](https://user-images.githubusercontent.com/1196252/109027576-3c750600-76c1-11eb-942d-6dc4ec452fe3.png)

Now if I want to create a page under `/Dave` it isn't obvious how.

![image](https://user-images.githubusercontent.com/1196252/109027850-82ca6500-76c1-11eb-813d-70911bdab46f.png)

You need to change `/new-page` at the bottom to `/Dave/daves-second-page`.

Once I finish creating that page, magically I have what I want.

![image](https://user-images.githubusercontent.com/1196252/109028257-e8b6ec80-76c1-11eb-9b84-9275a6836db3.png)

Now `/Dave` is no longer just a page it is **also** a virtual directory called "Dave".  Why?  Because it has subpages (pages that start with `/Dave/...`).  So when creating a new page I can select the "Dave" directory on the left and the page creation tool automatically fills in that prefix for me at the bottom.

![image](https://user-images.githubusercontent.com/1196252/109028591-3cc1d100-76c2-11eb-9d6d-d084b102f12c.png)

# Example 2: directory first

Instead of doing it as in Example 1, we can create the directory first (maybe more logical to you).

No `/Mary` virtual directory exists.  Create a new page at path `/Mary/marys-page` 

![image](https://user-images.githubusercontent.com/1196252/109029094-c7a2cb80-76c2-11eb-894a-f2ed5eaa6206.png)

and magically the `/Mary` virtual directory shows up.  There is still no `/Mary` page but you can create one now if you want.

![image](https://user-images.githubusercontent.com/1196252/109029692-616a7880-76c3-11eb-891b-89f5f174390f.png)

You can create further pages more easily now that the virtual directories `/Dave` and `/Mary` exist because you can select the directory in the page creating process and the path will be filled in for you.

# Summary

I agree it is a little clunky, especially for non-technical users.  But I think if you create the first page or two (for each of your virtual users)...to ensure the virtual directory shows up...I think you will be fine.  Making sure the virtual directory is already created is important so your non technical users can more intuitively add additional pages under that "directory".
